gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2455,42 +2721,81 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
     case OFTDate:
     {
         nType = NC_INT;
-        status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
-                            &nVarID);
+
+        if(!m_bLegacyCreateMode)
+        {
+            nVarID = m_poDS->vcdf.nc_def_vvar(pszVarName, nType, 1, &nMainDimId);
+        }
+
+        else
+        {
+            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
+                                &nVarID);
+        }
+
         NCDF_ERR(status);
         if( status != NC_NOERR )
         {
             return OGRERR_FAILURE;
         }
         nodata.nVal = NC_FILL_INT;
 
-        status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_UNITS,
+        if(!m_bLegacyCreateMode)
+        {
+            m_poDS->vcdf.nc_put_vatt_text(nVarID, CF_UNITS,
+                                 "days since 1970-1-1");
+        }
+
+        else
+        {
+            status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_UNITS,
                                  strlen("days since 1970-1-1"),
                                  "days since 1970-1-1");
-        NCDF_ERR(status);
+            NCDF_ERR(status);
+        }
 
         break;
     }
 
     case OFTDateTime:
     {
         nType = NC_DOUBLE;
-        status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
-                            &nVarID);
+
+        if(!m_bLegacyCreateMode)
+        {
+            nVarID = m_poDS->vcdf.nc_def_vvar(pszVarName, nType, 1, &nMainDimId);
+        }
+
+        else
+        {
+            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
+                                &nVarID);
+        }
+
         NCDF_ERR(status);
         if( status != NC_NOERR )
         {
             return OGRERR_FAILURE;
         }
         nodata.dfVal = NC_FILL_DOUBLE;
 
-        status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_UNITS,
+        if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation